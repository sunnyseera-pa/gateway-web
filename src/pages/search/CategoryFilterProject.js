import React, { Component } from 'react';
import { Row, Col, InputGroup, FormText } from 'react-bootstrap';

class CategoryFilterProject extends Component {
    state = {
        projectCategoriesSelected: [],
        projectCategoriesData: []
    };

    constructor(props) {
        super(props);
        this.state.projectCategoriesData = props.projectCategoriesData;
        this.state.projectCategoriesSelected = props.projectCategoriesSelected;
    }

    changeFilter = (e) => {
        const projectCategoriesSelected = this.state.projectCategoriesSelected
        let index

        if (e.target.checked) {
            projectCategoriesSelected.push(e.target.value)
        } else {
            index = projectCategoriesSelected.indexOf(e.target.value)
            projectCategoriesSelected.splice(index, 1)
        }

        this.setState({ projectCategoriesSelected: projectCategoriesSelected })
        this.props.updateOnFilter();
    }

    clearFilter = () => {
        const projectCategoriesSelected = this.state.projectCategoriesSelected;
        while (projectCategoriesSelected.length) { projectCategoriesSelected.pop(); }
        this.props.updateOnFilter();
    }

    render() {
        const { projectCategoriesData, projectCategoriesSelected } = this.state;

        if (!projectCategoriesData || projectCategoriesData.length === 0) {
            return (<></>);
        }

        return (
            <div>
                <div className="FilterCard mt-2">
                    <Row className="mt-2"  >
                        <Col xs={7} className="ml-3">
                            <span className="Gray800-14px-bold"> Category </span>
                            {projectCategoriesSelected.length === 0 ? <span /> :
                                <span> <div className="White-12px BubbleCounts"> {projectCategoriesSelected.length} </div> </span>
                            }
                        </Col>
                        <Col xs={3}>
                            {projectCategoriesSelected.length > 0 ?
                                <span>
                                    <button className="ClearButtons Purple-14px" onClick={() => this.clearFilter()}>
                                        Clear
                                    </button>
                                </span> : null}
                        </Col>
                    </Row>
                </div>
                <div className="AdFilters Gray800-14px">


                    <Row className="mb-3">
                        <Col xs={1}></Col>
                        <Col xs={11} className="ml-4">
                            {!projectCategoriesData ? '' : projectCategoriesData.map((category) => {
                                return <InputGroup >
                                    <InputGroup.Prepend>
                                        <InputGroup.Checkbox aria-label="Checkbox for following text input" name="toolCategory" checked={projectCategoriesSelected.indexOf(category) !== -1 ? "true" : ""} value={category} onChange={this.changeFilter} />
                                    </InputGroup.Prepend>
                                    <FormText className="Gray800-14px ml-4 mt-2 mb-2 pb-1" >{category}</FormText>
                                </InputGroup>
                            })}
                        </Col>
                    </Row>
                </div>
            </div>
        );
    }
}

export default CategoryFilterProject;