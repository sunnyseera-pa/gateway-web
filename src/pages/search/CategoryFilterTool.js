import React, { Component } from 'react';
import { Row, Col, InputGroup, FormText } from 'react-bootstrap';

class CategoryFilterTool extends Component {
    state = {
        toolCategoriesSelected: [],
        toolCategoriesData: []
    };

    constructor(props) {
        super(props);
        this.state.toolCategoriesData = props.toolCategoriesData;
        this.state.toolCategoriesSelected = props.toolCategoriesSelected;
    }

    changeFilter = (e) => {
        const toolCategoriesSelected = this.state.toolCategoriesSelected
        let index

        if (e.target.checked) {
            toolCategoriesSelected.push(e.target.value)
        } else {
            index = toolCategoriesSelected.indexOf(e.target.value)
            toolCategoriesSelected.splice(index, 1)
        }

        this.setState({ toolCategoriesSelected: toolCategoriesSelected })
        this.props.updateOnFilter();
    }

    clearFilter = () => {
        const toolCategoriesSelected = this.state.toolCategoriesSelected;
        while (toolCategoriesSelected.length) { toolCategoriesSelected.pop(); }
        this.props.updateOnFilter();
    }

    render() {
        const { toolCategoriesData, toolCategoriesSelected } = this.state;

        if (!toolCategoriesData || toolCategoriesData.length === 0) {
            return (<></>);
        }

        return (
            <div>
                <div className="FilterCard mt-2">
                    <Row className="mt-2"  >
                        <Col xs={7} className="ml-3">
                            <span className="Gray800-14px-bold"> Category </span>
                            {toolCategoriesSelected.length === 0 ? <span /> :
                                <span> <div className="White-12px BubbleCounts"> {toolCategoriesSelected.length} </div> </span>
                            }
                        </Col>
                        <Col xs={3}>
                            {toolCategoriesSelected.length > 0 ?
                                <span>
                                    <button className="ClearButtons Purple-14px" onClick={() => this.clearFilter()}>
                                        Clear
                                    </button>
                                </span> : null}
                        </Col>
                    </Row>
                </div>
                <div className="AdFilters Gray800-14px">


                    <Row className="mb-3">
                        <Col xs={1}></Col>
                        <Col xs={11} className="ml-4">
                            {!toolCategoriesData ? '' : toolCategoriesData.map((category) => {
                                return <InputGroup >
                                    <InputGroup.Prepend>
                                        <InputGroup.Checkbox aria-label="Checkbox for following text input" name="toolCategory" checked={toolCategoriesSelected.indexOf(category) !== -1 ? "true" : ""} value={category} onChange={this.changeFilter} />
                                    </InputGroup.Prepend>
                                    <FormText className="Gray800-14px ml-4 mt-2 mb-2 pb-1" >{category}</FormText>
                                </InputGroup>
                            })}
                        </Col>
                    </Row>
                </div>
            </div>
        );
    }
}

export default CategoryFilterTool;