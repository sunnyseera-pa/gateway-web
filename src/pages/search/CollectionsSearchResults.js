const CollectionsSearchResults = props => {
	const mapResults = React.useCallback(data => {
		return data.map(collection => {
			return (
				<Col sm={12} md={12} lg={6} style={{ 'text-align': '-webkit-center' }}>
					<CollectionCard key={collection.id} data={collection} />
				</Col>
			);
		});
	}, []);

	return <SearchResults {...props} results={mapResults} />;
};
