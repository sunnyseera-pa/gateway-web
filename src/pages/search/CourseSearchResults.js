const CourseSearchResults = props => {
	const mapResults = React.useCallback(data => {
		let courseRender = [];
		let currentHeader = '';

		return courseData.map(course => {
			if (!currentHeader && course.courseOptions.flexibleDates && currentHeader !== 'Flexible') {
				currentHeader = 'Flexible';
			} else if (
				!currentHeader &&
				course.courseOptions.startDate &&
				currentHeader !== moment(course.courseOptions.startDate).format('MMMM')
			) {
				currentHeader = moment(course.courseOptions.startDate).format('MMMM');
			}

			if (showHeader) {
				courseRender.push();
			}

			return (
				<>
					{currentHeader && (
						<Row className='courseDateHeader'>
							<Col>
								<span className='black-20-semibold'>{currentHeader}</span>
							</Col>
						</Row>
					)}
					<RelatedObject
						key={course.id}
						data={course}
						activeLink={true}
						onSearchPage={true}
						updateOnFilterBadge={this.updateOnFilterBadge}
					/>
				</>
			);
		});
	}, []);

	return <SearchResults {...props} results={mapResults} />;
};
