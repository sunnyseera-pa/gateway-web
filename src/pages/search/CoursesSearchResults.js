import React from 'react';
import { Col, Row } from 'react-bootstrap';
import RelatedObject from '../commonComponents/relatedObject/RelatedObject';
import SearchResults from '../commonComponents/SearchResults';
import moment from 'moment';

const CoursesSearchResults = ({ updateOnFilterBadge, ...outerProps }) => {
	const mapResults = React.useCallback(
		data => {
			let currentHeader = '';

			return data.map(course => {
				let output;

				if (!currentHeader) {
					if (course.courseOptions.flexibleDates && currentHeader !== 'Flexible') {
						currentHeader = 'Flexible';
					} else if (course.courseOptions.startDate && currentHeader !== moment(course.courseOptions.startDate).format('MMMM')) {
						currentHeader = moment(course.courseOptions.startDate).format('MMMM');
					}

					output = (
						<Row className='courseDateHeader'>
							<Col>
								<span className='black-20-semibold'>{currentHeader}</span>
							</Col>
						</Row>
					);
				}

				return (
					<>
						{output}
						<RelatedObject key={course.id} data={course} activeLink={true} onSearchPage={true} updateOnFilterBadge={updateOnFilterBadge} />
					</>
				);
			});
		},
		[updateOnFilterBadge]
	);

	return <SearchResults {...outerProps} results={mapResults} />;
};

export default CoursesSearchResults;
