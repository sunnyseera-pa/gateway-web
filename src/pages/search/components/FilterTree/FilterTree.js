/** @jsx jsx */
import { jsx } from '@emotion/react';
import { flatMapDeep } from 'lodash';
import React from 'react';
import { Accordion } from 'react-bootstrap';
import CheckboxTree from '../../../../components/CheckboxTree';
import TreeSubHeader from '../TreeSubHeader';
import * as styles from './FilterTree.styles';

const FilterTree = ({ node, filters, checked, expanded, onCheck }) => {
	const [nodesChecked, setNodesChecked] = React.useState(checked);
	const [nodesExpanded, setNodesExpanded] = React.useState(expanded);

	React.useEffect(() => {
		setNodesChecked(checked);
	}, [checked]);

	const flattenObject = (filters, nodes = []) => {
		filters.forEach(filter => {
			nodes.push(filter);

			flattenObject(filter.children, nodes);
		});

		return nodes;
	};

	const handleChecked = React.useCallback(
		checked => {
			setNodesChecked(checked);

			const nodes = flattenObject(filters).filter(filter => {
				return checked.includes(filter.value);
			});

			if (onCheck) onCheck(nodes, node.key, true);
		},
		[node]
	);
	return (
		<Accordion css={styles.root}>
			<Accordion.Toggle eventKey='0' css={styles.toggle}>
				<TreeSubHeader node={node} />
			</Accordion.Toggle>
			<Accordion.Collapse eventKey='0'>
				<CheckboxTree nodes={filters} checked={nodesChecked} expanded={nodesExpanded} onCheck={handleChecked} onExpand={setNodesExpanded} />
			</Accordion.Collapse>
		</Accordion>
	);
};

export default FilterTree;
