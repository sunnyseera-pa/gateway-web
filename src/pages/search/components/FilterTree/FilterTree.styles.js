import { css } from '@emotion/react';

export const root = css`
	.ui-CheckboxTree .react-checkbox-tree {
		font-size: 13px;
	}
`;

export const toggle = css`
	background: none;
	border: none;
	outline: 0;
	padding: 0;

	&:focus {
		outline: 0;
	}
`;
