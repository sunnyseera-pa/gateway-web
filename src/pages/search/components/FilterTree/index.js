import FilterTree from './FilterTree';

export default FilterTree;
