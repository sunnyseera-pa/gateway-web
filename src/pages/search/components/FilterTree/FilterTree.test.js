import { render, waitFor } from '@testing-library/react';
import React from 'react';
import FilterTree from '.';
import { findAllByKey } from '../../../../utils/GeneralHelper.util';
import { mockDatasetFilters } from '../../../../services/search/mockMsw';

const spatialNode = findAllByKey(mockDatasetFilters, (key, value) => {
	return key === 'key' && value === 'spatial';
})[0];

const props = {
	node: spatialNode,
	filters: spatialNode.filtersv2,
	highlighted: ['UK', 'UK, Wales'],
	checked: ['UK', 'UK, Wales'],
	expanded: ['UK', 'UK, Wales'],
	onCheck: jest.fn(),
};

let wrapper;

describe('Give a FilterTree component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<FilterTree {...props} />, {
				wrapper: Providers,
			});
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then has the correct nodes highlighted', () => {
			expect(wrapper.getByText('UK')).toBeTruthy();
			expect(wrapper.getByText('Wales')).toBeTruthy();
		});

		it('Then has the correct nodes not highlighted', () => {
			expect(wrapper.queryByText('India').classList[0]).toEqual('checkbox-text');
			expect(wrapper.queryByText('International').classList[0]).toEqual('checkbox-text');
			expect(wrapper.queryByText('Worldwide').classList[0]).toEqual('checkbox-text');
		});

		describe('And the label is clicked', () => {
			it('Then has called onCheck', async () => {
				const label = wrapper.container.querySelectorAll('label')[0];

				fireEvent.click(label, { target: { checked: true } });

				await waitFor(() => expect(props.onCheck).toHaveBeenCalled());
			});
		});

		describe('And it is searched', () => {
			beforeAll(() => {
				wrapper.rerender(<FilterTree {...props} searchValue='Wales' />, {
					wrapper: Providers,
				});
			});

			it('Then has the correct nodes highlighted', () => {
				expect(wrapper.getByText('UK')).toBeTruthy();
				expect(wrapper.getByText('Wales')).toBeTruthy();
			});

			it('Then has the correct nodes not highlighted', () => {
				expect(wrapper.queryByText('India')).toBeFalsy();
				expect(wrapper.queryByText('International')).toBeFalsy();
				expect(wrapper.queryByText('Worldwide')).toBeFalsy();
			});
		});
	});
});
