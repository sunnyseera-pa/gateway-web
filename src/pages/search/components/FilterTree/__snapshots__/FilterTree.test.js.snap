// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Give a FilterTree component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="css-1fe7u86 ui-CheckboxTree css-4kcmtq-CheckboxTree"
  >
    <div
      class="react-checkbox-tree rct-icons-fa4"
      id="rct-ys35LVU8x3shv-TS6g57v"
    >
      <ol>
        <li
          class="rct-node rct-node-parent rct-node-expanded"
        >
          <span
            class="rct-text"
          >
            <button
              aria-label="Toggle"
              class="rct-collapse rct-collapse-btn"
              title="Toggle"
              type="button"
            />
            <label
              for="rct-ys35LVU8x3shv-TS6g57v-UK"
            >
              <input
                checked=""
                id="rct-ys35LVU8x3shv-TS6g57v-UK"
                type="checkbox"
              />
              <span
                aria-checked="true"
                aria-disabled="false"
                class="rct-checkbox"
                role="checkbox"
                tabindex="0"
              >
                <span
                  class="rct-icon rct-icon-check"
                />
              </span>
              <span
                class="rct-node-icon"
              />
              <span
                class="rct-title"
              >
                UK
              </span>
            </label>
          </span>
          <ol>
            <li
              class="rct-node rct-node-leaf"
            >
              <span
                class="rct-text"
              >
                <span
                  class="rct-collapse"
                >
                  <span
                    class="rct-icon"
                  />
                </span>
                <label
                  for="rct-ys35LVU8x3shv-TS6g57v-UK,_Wales"
                >
                  <input
                    checked=""
                    id="rct-ys35LVU8x3shv-TS6g57v-UK,_Wales"
                    type="checkbox"
                  />
                  <span
                    aria-checked="true"
                    aria-disabled="false"
                    class="rct-checkbox"
                    role="checkbox"
                    tabindex="0"
                  >
                    <span
                      class="rct-icon rct-icon-check"
                    />
                  </span>
                  <span
                    class="rct-node-icon"
                  />
                  <span
                    class="rct-title"
                  >
                    Wales
                  </span>
                </label>
              </span>
            </li>
          </ol>
        </li>
        <li
          class="rct-node rct-node-leaf"
        >
          <span
            class="rct-text"
          >
            <span
              class="rct-collapse"
            >
              <span
                class="rct-icon"
              />
            </span>
            <label
              for="rct-ys35LVU8x3shv-TS6g57v-India"
            >
              <input
                id="rct-ys35LVU8x3shv-TS6g57v-India"
                type="checkbox"
              />
              <span
                aria-checked="false"
                aria-disabled="false"
                class="rct-checkbox"
                role="checkbox"
                tabindex="0"
              >
                <span
                  class="rct-icon rct-icon-uncheck"
                />
              </span>
              <span
                class="rct-node-icon"
              />
              <span
                class="rct-title"
              >
                <span
                  class="checkbox-text"
                >
                  India
                </span>
              </span>
            </label>
          </span>
        </li>
        <li
          class="rct-node rct-node-leaf"
        >
          <span
            class="rct-text"
          >
            <span
              class="rct-collapse"
            >
              <span
                class="rct-icon"
              />
            </span>
            <label
              for="rct-ys35LVU8x3shv-TS6g57v-International"
            >
              <input
                id="rct-ys35LVU8x3shv-TS6g57v-International"
                type="checkbox"
              />
              <span
                aria-checked="false"
                aria-disabled="false"
                class="rct-checkbox"
                role="checkbox"
                tabindex="0"
              >
                <span
                  class="rct-icon rct-icon-uncheck"
                />
              </span>
              <span
                class="rct-node-icon"
              />
              <span
                class="rct-title"
              >
                <span
                  class="checkbox-text"
                >
                  International
                </span>
              </span>
            </label>
          </span>
        </li>
        <li
          class="rct-node rct-node-leaf"
        >
          <span
            class="rct-text"
          >
            <span
              class="rct-collapse"
            >
              <span
                class="rct-icon"
              />
            </span>
            <label
              for="rct-ys35LVU8x3shv-TS6g57v-Worldwide"
            >
              <input
                id="rct-ys35LVU8x3shv-TS6g57v-Worldwide"
                type="checkbox"
              />
              <span
                aria-checked="false"
                aria-disabled="false"
                class="rct-checkbox"
                role="checkbox"
                tabindex="0"
              >
                <span
                  class="rct-icon rct-icon-uncheck"
                />
              </span>
              <span
                class="rct-node-icon"
              />
              <span
                class="rct-title"
              >
                <span
                  class="checkbox-text"
                >
                  Worldwide
                </span>
              </span>
            </label>
          </span>
        </li>
      </ol>
    </div>
  </div>
</div>
`;
